Src/ILGPU.Algorithms/Runtime/Cuda/API/CuFFTAPI.cs

```diff
@@ -0,0 +1,672 @@
+            int[] inembed,
+            int istride,
+            int idist,
+            int[] onembed,
+            int ostride,
+            int odist,
+            CuFFTType type,
+            int batch,
+            UIntPtr[] workSize)
+        {
+            fixed (int* nPtr = n)
+            fixed (int* inembedPtr = inembed)
+            fixed (int* onembedPtr = onembed)
+            fixed (UIntPtr* workSizePtr = workSize)
+            {
+                return MakePlanMany(
+                    plan,
+                    rank,
+                    nPtr,
+                    inembedPtr,
+                    istride,
+                    idist,
+                    onembedPtr,
+                    ostride,
+                    odist,
+                    type,
+                    batch,
+                    workSizePtr);
+            }
+        }
+
+        /// <summary>
+        /// Configures a custom plan.
+        /// </summary>
+        /// <param name="plan">The plan handle.</param>
+        /// <param name="rank">The transform.</param>
+        /// <param name="n">The transform dimensions.</param>
+        /// <param name="inembed">The storage dimensions of the input data.</param>
+        /// <param name="istride">The stride of the input data.</param>
+        /// <param name="idist">The distance of the input data.</param>
+        /// <param name="onembed">The storage dimensions of the output data.</param>
+        /// <param name="ostride">The stride of the output data.</param>
+        /// <param name="odist">The distance of the output data.</param>
+        /// <param name="type">The transform type.</param>
+        /// <param name="batch">The number of transforms.</param>
+        /// <param name="workSize">The work size.</param>
+        /// <returns>The error code.</returns>
+        public unsafe CuFFTResult MakePlanMany(
+            IntPtr plan,
+            int rank,
+            long[] n,
+            long[] inembed,
+            long istride,
+            long idist,
+            long[] onembed,
+            long ostride,
+            long odist,
+            CuFFTType type,
+            long batch,
+            UIntPtr[] workSize)
+        {
+            fixed (long* nPtr = n)
+            fixed (long* inembedPtr = inembed)
+            fixed (long* onembedPtr = onembed)
+            fixed (UIntPtr* workSizePtr = workSize)
+            {
+                return MakePlanMany64(
+                    plan,
+                    rank,
+                    nPtr,
+                    inembedPtr,
+                    istride,
+                    idist,
+                    onembedPtr,
+                    ostride,
+                    odist,
+                    type,
+                    batch,
+                    workSizePtr);
+            }
+        }
+
+        #endregion
+
+        #region Refined Estimated Size Of Work Area
+
+        /// <summary>
+        /// Estimates the work area for a 1D plan.
```

Comment:
Question: is this the right description?

Reply:
I have updated the description. It does provide an estimate, but it is more accurate because the caller also needs to pass in a plan.